src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,327 @@
+            grabbedEdges.right = false;
+          }
+
+          if (!this.boxVisible() ||
+              (!grabbedEdges.top && !grabbedEdges.bottom &&
+               !grabbedEdges.left && !grabbedEdges.right)
+             ) {
+            this.bounds({
+              topLeft: s,
+              bottomRight: s
+            });
+            // a new box is effectively "grabbed" at the bottom-right corner
+            grabbedEdges.bottom = true;
+            grabbedEdges.right = true;
+          }
+
+          this.boxVisible(true);
+          var bounds = this.bounds();
+          // copy points so changes to topLeft and bottomRight don't mutate bounds
+          topLeft = { x: bounds.topLeft.x, y: bounds.topLeft.y };
+          bottomRight = { x: bounds.bottomRight.x, y: bounds.bottomRight.y };
+          if (this._dragStartCallback) {
+            this._dragStartCallback(bounds);
+          }
+        });
+
+        this._dragInteraction.onDrag((s: Point, e: Point) => {
+          if (grabbedEdges.bottom) {
+            bottomRight.y = e.y;
+          } else if (grabbedEdges.top) {
+            topLeft.y = e.y;
+          }
+
+          if (grabbedEdges.right) {
+            bottomRight.x = e.x;
+          } else if (grabbedEdges.left) {
+            topLeft.x = e.x;
+          }
+
+          this.bounds({
+            topLeft: topLeft,
+            bottomRight: bottomRight
+          });
+
+          if (this._dragCallback) {
+            this._dragCallback(this.bounds());
+          }
+        });
+
+        this._dragInteraction.onDragEnd((s: Point, e: Point) => {
+          if (s.x === e.x && s.y === e.y) {
+            this.boxVisible(false);
+          }
+
+          if (this._dragEndCallback) {
+            this._dragEndCallback(this.bounds());
+          }
+        });
+      }
+
+      protected _setup() {
+        super._setup();
+
+        var createLine = () => this._box.append("line").style({
+                                 "opacity": 0,
+                                 "stroke": "pink"
+                               });
+        this._detectionEdgeT = createLine().classed("drag-edge-tb", true);
+        this._detectionEdgeB = createLine().classed("drag-edge-tb", true);
+        this._detectionEdgeL = createLine().classed("drag-edge-lr", true);
+        this._detectionEdgeR = createLine().classed("drag-edge-lr", true);
+
+        var createCorner = () => this._box.append("circle")
+                                     .style({
+                                       "opacity": 0,
+                                       "fill": "pink"
+                                     });
+        this._detectionCornerTL = createCorner().classed("drag-corner-tl", true);
+        this._detectionCornerTR = createCorner().classed("drag-corner-tr", true);
+        this._detectionCornerBL = createCorner().classed("drag-corner-bl", true);
+        this._detectionCornerBR = createCorner().classed("drag-corner-br", true);
```

Comment:
No, because then when someone sets the properties to `true` we have to go create the elements. If the property is `false` the presence of the elements doesn't do anything bad.